src/Domain/Services/ExtendRestApi.php

```diff
@@ -135,6 +142,60 @@ public function get_endpoint_data( $endpoint, array $passed_args = [] ) {
 		return (object) $registered_data;
 	}
 
+	/**
+	 * Registers and validates payment requirements callbacks.
+	 *
+	 * @param array $args {
+	 *     Array of registration data.
+	 *
+	 *     @type callable $data_callback Callback executed to add payment requirements data.
+	 * }
+	 *
+	 * @throws Exception On failure to register.
+	 * @return boolean True on success.
+	 */
+	public function register_payment_requirements( $args ) {
+		if ( ! is_callable( $args['data_callback'] ) ) {
+			$this->throw_exception( '$data_callback must be a callable function.' );
+		}
+
+		$this->payment_requirements[] = $args['data_callback'];
```

Comment:
Why are we using callbacks if we're not passing any variables, for me it would make more sense for this to accept a string or an array of strings.

Reply:
We are using callbacks because we want the code to be only executed if it is needed. This way we only fetch the values when we prepare the response. That is what I understand from your tips:
> Lazy loaded, only executing the developer code once we need it.
Besides, it is nice that the signature for this is similar to the other extension point that we have. It makes things less confusing.

Reply:
I see your point, in my mind, adding a requirement is a static task, but it actually requires some logic and queries on the current state of things (cart, order, settings...) so a callback is good.